library/src/main/java/in/krharsh17/barview/BarView.java

```diff
@@ -298,6 +298,23 @@ public static String getRandomColor() {
         return color.toString();
     }
 
+    /**
+     * This function returns a random color based on a constant {@value #CHAR_ARRAY}
+     * It modifies the passed in hex value by randomly choosing between any of the 16
+     * literals in the char array and replacing them from index 4 till 6 in the passed in
+     * literal of the format #XXXXXX
+     *
+     * @param preciseColor is the color user wants the bar to be precisely like
```

Comment:
Actually, the user is going to specify an approximate colour for reference. If a precise colour was needed, then there was no point in randomizing the colour bits.

Reply:
I mean user will give a color and we will flip the 2 most insignificant bits to ensure that it stays around that color

Reply:
Yes, so what I meant was that the user does not want the colour of the bar to be exactly the one he has entered, so I think the name `preciseColor` can be misleading.

Reply:
Okay Ill change that to approximateColor 👍